README.md

```diff
@@ -1,14 +1,77 @@
 # Synta
 
 Synta [^1] è uno strumento a linea di comando per svolgere svariate operazioni
-sul formato di definizione regolare usato internamente a [csunibo](https://github.com/csunibo)
-per vincolare le nomenclature dei file. Con Synta è possibile:
-- Verificare la validità di un file `.synta`
-- Convertire un file `.synta` in una espressione regolare
-- Convertire un file `.synta` in un file JSON che descrivre i contenuti dei file.
-    Il file JSON generato è descritto dallo schema JSON disponibile [qui](TODO).
+sul formato di definizione regolare usato internamente a
+[csunibo](https://github.com/csunibo) per vincolare le nomenclature dei file.
+Con Synta è possibile:
+
+- verificare la validità di un file `.synta`;
+- convertire un file `.synta` in una espressione regolare;
+- convertire un file `.synta` in un file JSON che descrive i contenuti dei file.
+  Il file JSON generato è descritto dallo schema JSON disponibile [qui](TODO).
+
+# Definizione del linguaggio
+
+Il linguaggio Synta è definito dalla seguente BNF:
+
+```bnf
+<upperletter> ::= "A" | "B" | "C" | "D" | "E" | "F" | "G" | "H" | "I" | "J" |
+                  "K" | "L" | "M" | "N" | "O" | "P" | "Q" | "R" | "S" | "T" |
+                  "U" | "V" | "W" | "X" | "Y" | "Z"
+<lowerletter> ::= "a" | "b" | "c" | "d" | "e" | "f" | "g" | "h" | "i" | "j" |
+                  "k" | "l" | "m" | "n" | "o" | "p" | "q" | "r" | "s" | "t" |
+                  "u" | "v" | "w" | "x" | "y" | "z"
+<letter>      ::= <lowerletter> | <upperletter> | ":" | ";" | "," | "." | "-" | "_"
```

Comment:
La punteggiatura è una lettera?

Reply:
Se hai un nome migliore cambio!

Reply:
Beh word mi sembra inutilizzata, e quindi si può rimuovere insieme a letter. Se invece servono a qualcosa e me lo sono perso in base al loro scopo ripensiamo al nome.

Reply:
Oddio non sono usate nei commenti?

Reply:
Si, nei commenti ho messo `alphanum` ma volevo mettere `word`